pkg/reconciler/contour/resources/httpproxy.go

```diff
@@ -129,22 +108,6 @@ func MakeHTTPProxies(ctx context.Context, ing *v1alpha1.Ingress, serviceToProtoc
 				Idle:     config.FromContext(ctx).Contour.TimeoutPolicyIdle,
 			}
 
-			// By default retry on connection problems twice.
-			// This matches the default behavior of Istio:
-			// https://istio.io/latest/docs/concepts/traffic-management/#retries
-			// However, in addition to the codes specified by istio
-			retry := defaultRetryPolicy()
-			if path.DeprecatedRetries != nil && path.DeprecatedRetries.Attempts > 0 {
-				retry.NumRetries = int64(path.DeprecatedRetries.Attempts)
-
-				// When retries is specified explicitly, then we retry some http-level failures as well.
-				retry.RetryOn = append(retry.RetryOn, "5xx")
-
-				if path.DeprecatedRetries.PerTryTimeout != nil {
-					retry.PerTryTimeout = path.DeprecatedRetries.PerTryTimeout.Duration.String()
-				}
```

Comment:
By dropping the default you are seriously weakening things, so I'd like to better understand what you are after.

Reply:
I'd think retrying connection level stuff seems fine, so should we drop only the 5xx retries?

@mattmoor for context: @duglin hit an issue where Istio was still exercising retries, though we thought that we didn't use retries at all anymore in Knative.

Reply:
As long as the retires don't result in duplicate events being delivered (attempted from the ksvc point of view) then retrying connection errors (but not http errors) might be ok.

Reply:
the default retry policy contour uses is an extension of the retry policy Istio uses when nothing is specified.  5xx was something we added, which is what the `DeprecatedRetries` field is about, so if this is about dropping that field then 👍 , but some of the code here needs to stay.

Reply:
Thank you. I updated as:
- Drop `DeprecatedRetries` handling.
- Drop retry for 5xx from defaultRetryPolicy but keep retry for the connection error.